src/core/stdc/stdlib.d

```diff
@@ -128,13 +128,6 @@ version (CRuntime_Microsoft)
         return strtod(nptr, endptr);
     }
 }
-else version (MinGW)
-{
-    ///
-    real __mingw_strtold(scope inout(char)* nptr, scope inout(char)** endptr);
-    ///
-    alias __mingw_strtold strtold;
-}
```

Comment:
Is this intentional? I.e., leaving the specialized MinGW printf functions to support full x87 precision, but crippling strtold to double precision?

Reply:
That was an oversight, thanks.

Many MinGW specific code paths in phobos (especially in unit tests) are caused by different formatting of reals caused by the `_mingw_printf` functions. I'd like to get rid of this. OTOH we probably can't default to 64 bit reals, as people expect MinGW gcc to be compatible with MinGW gdc with default settings. Long-term I guess I'll track down where in phobos we call the C snprintfs, convert real to double there instead and make it call the standard MSVC snprintfs. Not sure what to do for these C bindings then though, ideally users should be able to call either the extended mingw function or the default msvc version. 

But all that is probably for GCC 10, I'd be happy if we get some sort of even basic MinGW support into GCC9, we're actually already quite late for that.

Reply:
> Many MinGW specific code paths in phobos (especially in unit tests) are caused by different formatting of reals caused by the _mingw_printf functions.

I know, and I don't like it either, but it's just due to the weird hybrid nature of the MinGW runtime - MSVC runtime + 80-bit long-double + some C99/POSIX stuff which used to be missing in earlier MSVC versions (but C99 support is thankfully nearly complete nowadays).

The real shame to be fixed IMO is that Phobos still depends on C formatting functions.

Reply:
> The real shame to be fixed IMO is that Phobos still depends on C formatting functions.

I agree.